### PR TITLE
threading macro example

### PR DESCRIPTION
**Pull request by francescoagati**
an implementation of clojure threading macro in haxe

### CODE REVIEW

File: assets/content/cookbook/Macros/threading-macro.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    var res = _3;
+    console.log(res);
+};
+Main.main();
+})();
+```
+
+## The code generated with the static analyzer
+
+In this case thanks to the static analyzer the temp variables
```
Comment (markknol) [inline assets/content/cookbook/Macros/threading-macro.md]: This paragraph feels incomplete.

File: assets/content/cookbook/Macros/threading-macro.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+};
+Main.main = function() {
+	console.log(Main.sum(3,Main.sum(Main.sum(Main.sum(0,1),1),2)));
+};
+Main.main();
+})();
+```
+
+## With inline function 
+with the code inlined of the function
```
Comment (markknol) [inline assets/content/cookbook/Macros/threading-macro.md]: This paragraph feels incomplete too.

File: assets/content/cookbook/Macros/threading-macro.md (hunk 1 above)
Comment (francescoagati) [reply assets/content/cookbook/Macros/threading-macro.md]: better

File: assets/content/cookbook/Macros/threading-macro.md (hunk 2 above)
Comment (francescoagati) [reply assets/content/cookbook/Macros/threading-macro.md]: better

**Comment by francescoagati** [comment]
the modification i have do are good now?

**Comment by markknol** [comment]
I think its all good now, thanks for the contribution!

**Comment by francescoagati** [comment]
thanks to you

**Pull request merged**